internal/providers/client/databricks_client.go

```diff
@@ -0,0 +1,67 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go/client"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/databricks/terraform-provider-databricks/commands"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// PrepareDatabricksClient makes some common adjustments to the config that apply in all cases
+// and returns a ready-to-use Databricks client. This includes:
+// - mapping deprecated auth types to their newer counterparts
+// - ensuring the config is resolved
+// - setting a default retry timeout if not set
+// - setting a default HTTP timeout if not set
+//
+// TODO: this should be colocated with the definition of DatabricksClient in common/client.go, but
+// this isn't possible without introducing a circular dependency. Fixing this will require refactoring
+// DatabricksClient out of the common package.
+func PrepareDatabricksClient(ctx context.Context, cfg *config.Config, configCustomizer func(*config.Config) error) (*common.DatabricksClient, error) {
+	if cfg.AuthType != "" {
+		// mapping from previous Google authentication types
+		// and current authentication types from Databricks Go SDK
+		oldToNewerAuthType := map[string]string{
+			"google-creds":     "google-credentials",
+			"google-accounts":  "google-id",
+			"google-workspace": "google-id",
+		}
+		newer, ok := oldToNewerAuthType[cfg.AuthType]
+		if ok {
+			tflog.Info(ctx, fmt.Sprintf("Changing required auth_type from %s to %s", cfg.AuthType, newer))
```

Comment:
Is there any value in specifying sdkv2/pluginfw here? If so, maybe our tflog could be parameterized according to the provider?

Reply:
This is just to distinguish the log message printed from the plugin framework provider implementation vs the SDKv2 implementation. The same log message is printed on both sides, this way we can tell which log message came from which implementation. We could do something like this with the `additionalFields` parameter of the log message, but it is about the same.